core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -74,7 +75,7 @@
     public static final String SAFE_PARAMETERS_SYSTEM_PROPERTY_NAME = ParametersAction.class.getName() +
             ".safeParameters";
 
-    private transient List<String> safeParameters;
+    private Set<String> safeParameters;
```

Comment:
> Then he installs 2.0 to get a missing feature

but features in 2.0 will by virtue be in 2.5.1..... and this is a downgrade which is never guaranteed to work.

So maybe if you installed 2.5.1 and then upgraded to 2.6...

But in either case I don't think you will get `OldDataMonitor` issues as the data can be successfully restored without exception as the field is still there - so can be restored.